Source/ASNetworkImageNode.mm

```diff
@@ -322,13 +349,15 @@ - (void)displayWillStartAsynchronously:(BOOL)asynchronously
   if (asynchronously == NO && _cacheFlags.cacheSupportsSynchronousFetch) {
     ASDN::MutexLocker l(__instanceLock__);
 
-    if (_imageLoaded == NO && _URL && _downloadIdentifier == nil) {
-      UIImage *result = [[_cache synchronouslyFetchedCachedImageWithURL:_URL] asdk_image];
-      if (result) {
-        [self _locked_setCurrentImageQuality:1.0];
-        [self _locked__setImage:result];
-        
-        _imageLoaded = YES;
+    if (_imageLoaded == NO && _URLs.count > 0 && _downloadIdentifier == nil) {
+      for (NSURL *url in _URLs) {
+        UIImage *result = [[_cache synchronouslyFetchedCachedImageWithURL:url] asdk_image];
+        if (result) {
+          [self _locked_setCurrentImageQuality:1.0];
+          [self _locked__setImage:result];
+          
+          _imageLoaded = YES;
+        }
```

Comment:
Looks like we're fetching images from low cost to higher cost and don't bail as soon as there is a cache hit. Does this mean that if there are multiple images, we'll end up fetching all of them but only use the best one? If that's the case, should we instead fetch in reverse order and stop as early as possible?

Reply:
Yeah, good catch!